File: helfomat-infrastructure-config/src/main/java/de/helfenkannjeder/helfomat/infrastructure/config/ConfigFileLoaderConfig.java
package de.helfenkannjeder.helfomat.infrastructure.config;

import org.springframework.beans.factory.config.YamlPropertiesFactoryBean;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.support.PropertySourcesPlaceholderConfigurer;
import org.springframework.core.io.ClassPathResource;

import java.util.Objects;

/**
 * @author Valentin Zickner
 */
@Configuration
public class ConfigFileLoaderConfig {

    @Bean
    public static PropertySourcesPlaceholderConfigurer properties() {
        PropertySourcesPlaceholderConfigurer propertySourcesPlaceholderConfigurer = new PropertySourcesPlaceholderConfigurer();
        YamlPropertiesFactoryBean yaml = new YamlPropertiesFactoryBean();
        yaml.setResources(new ClassPathResource("config-repository-defaults.yml"));
        propertySourcesPlaceholderConfigurer.setProperties(Objects.requireNonNull(yaml.getObject()));
        return propertySourcesPlaceholderConfigurer;
    }

}

File: helfomat-infrastructure-config/src/main/kotlin/de/helfenkannjeder/helfomat/infrastructure/config/ConfigFileLoaderConfig.kt
package de.helfenkannjeder.helfomat.infrastructure.config

import org.springframework.beans.factory.config.YamlPropertiesFactoryBean
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.support.PropertySourcesPlaceholderConfigurer
import org.springframework.core.io.ClassPathResource

/**
 * @author Valentin Zickner
 */
@Configuration
open class ConfigFileLoaderConfig {

    @Bean
    open fun properties(): PropertySourcesPlaceholderConfigurer {
        val propertySourcesPlaceholderConfigurer = PropertySourcesPlaceholderConfigurer()
        val yaml = YamlPropertiesFactoryBean()
        yaml.setResources(ClassPathResource("config-repository-defaults.yml"))
        val properties = yaml.getObject()
        if (properties != null) {
            propertySourcesPlaceholderConfigurer.setProperties(properties)
        }
        return propertySourcesPlaceholderConfigurer
    }

}